internal/controllers/nmc_reconciler.go

```diff
@@ -127,8 +127,19 @@ func (r *NMCReconciler) Reconcile(ctx context.Context, req reconcile.Request) (r
 		return ctrl.Result{}, fmt.Errorf("could not get node %s: %v", nmcObj.Name, err)
 	}
 
-	// skipping handling NMC spec, labelling, events until node becomes ready
+	// skipping handling NMC spec, events until node becomes ready
+	// removing label of loaded kmods
 	if !r.nodeAPI.IsNodeSchedulable(&node) {
+		var labelsToRemove []string
```

Comment:
Tomer, wdyt about moving this whole if section to a nodeAPI? This code does not require any data from nmc, it just looks at the node.

Reply:
+1

Reply:
sure